src/ess/amor/utils.py

```diff
@@ -55,4 +65,29 @@ def theta_grid(
     return grid
 
 
-providers = (theta_grid,)
+def qgrid(
+    detector_rotation: DetectorRotation[SampleRun],
+    sample_rotation: SampleRotation[SampleRun],
+    wbins: WavelengthBins,
+    bdlims: BeamDivergenceLimits,
+) -> QBins:
+    '''Generates a suitable Q-binnning from
+    the limits on wavelength and divergence angle.'''
+    theta_min = (
+        bdlims[0].to(unit='rad', copy=False)
+        + detector_rotation.to(unit='rad', dtype='float64')
+        - sample_rotation.to(unit='rad', dtype='float64')
+    )
+    theta_max = (
+        bdlims[-1].to(unit='rad', copy=False)
+        + detector_rotation.to(unit='rad', dtype='float64')
+        - sample_rotation.to(unit='rad', dtype='float64')
+    )
+    wmin, wmax = wbins[0], wbins[-1]
+    qmin = reflectometry_q(wavelength=wmax, theta=theta_min)
+    qmax = reflectometry_q(wavelength=wmin, theta=theta_max)
+    qmin = max(qmin, sc.scalar(1e-3, unit='1/angstrom'))
```

Comment:
Any reason for this particular hard maximum? Should be documented in the docstring at least.